### PR TITLE
docs: update `template` for Nomad Variables

### PR DESCRIPTION
**Pull request by tgross**
Preview: https://nomad-gsgqqf402-hashicorp.vercel.app/docs/job-specification/template#nomad-variables

Once https://github.com/hashicorp/consul-template/pull/1632 is merged (after Nomad 1.4.0 beta goes out), we'll probably want to include a link to the docs over there as well.

### CODE REVIEW

File: website/content/docs/job-specification/template.mdx [hunk 1]
```diff
@@ -365,6 +365,179 @@ EOH
...
+}
+EOH
+}
+```
+
+The `nomadVarListSafe` function works identically to `nomadVarList`, but refuses
+to render the template if the variable list query returns blank/empty data.
+
+
+#### `nomadVar` and `nomadVarSafe`
```
Comment (angrycub) [inline website/content/docs/job-specification/template.mdx:449]: No nomadVarSafe
```suggestion
#### `nomadVar`
```

File: website/content/docs/job-specification/template.mdx [hunk 2]
```diff
@@ -365,6 +365,179 @@ EOH
...
+```hcl
+template {
+  data        = <<EOH
+{{ with nomadVar "nomad/jobs/redis@example_namespace" }}{{ .maxconns }}{{ end }}
+EOH
+}
+```
+
+The `nomadVarSafe` function works identically to `nomadVar`, but refuses to
+render the template if the variable query returns blank/empty data.
```
Comment (angrycub) [inline website/content/docs/job-specification/template.mdx:517]:
```suggestion
```

**Review by angrycub — APPROVED** [review]
Some corrections to be done, but no need to rereview after.

File: website/content/docs/job-specification/template.mdx (hunk 1 above)
Comment (tgross) [reply website/content/docs/job-specification/template.mdx]: Oh! Removed from the first paragraph as well.

**Pull request merged**